addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1321,19 +1321,29 @@ const Wysiwyg = Widget.extend({
         }
 
         if (params.node) {
-            params.node.replaceWith(element);
+            const changedImage = params.node.tagName === element.tagName && element.tagName === 'IMG';
+            if (changedImage) {
+                // Instead of replacing an image with another one and
+                // recreating its snippet editor, its attributes are
+                // changed, and the snippet editor is kept.
+                for (const attribute of params.node.attributes) {
+                    params.node.setAttribute(attribute.nodeName, '');
```

Comment:
Should not you use `removeAttribute` ?

Reply:
Yes... It's even recommended by the doc https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute#usage_notes
Changed to use removeAttribute